Tests/tests/test_OCIContainer.cpp

```diff
@@ -0,0 +1,211 @@
+using ::testing::NiceMock;
+
+class OCIContainerTest : public ::testing::Test {
+protected:
+    Core::ProxyType<Plugin::OCIContainer> plugin;
+    Core::JSONRPC::Handler& handler;
+    Core::JSONRPC::Connection connection;
+    string response;
+
+    OCIContainerTest()
+        : plugin(Core::ProxyType<Plugin::OCIContainer>::Create())
+        , handler(*(plugin))
+        , connection(1, 0)
+        {
+        }
+        virtual ~OCIContainerTest() = default;
+};
+
+class OCIContainerInitializedTest : public OCIContainerTest {
+protected:
+    NiceMock<ServiceMock> service;
+    DobbyProxyMock dobbymock;
+    IpcServiceMock ipcservicemock;
+
+    OCIContainerInitializedTest()
+        : OCIContainerTest()
+    {
+        DobbyProxy::getInstance().impl = &dobbymock;
+        IpcService::getInstance().impl = &ipcservicemock;
+
+        EXPECT_CALL(ipcservicemock, start())
+            .WillOnce(::testing::Return(true));
+
+        EXPECT_CALL(dobbymock, registerListener(::testing::_, ::testing::_))
+            .WillOnce(::testing::Return(5));
+
+        EXPECT_EQ(string(""), plugin->Initialize(&service));
+    }
+
+    virtual ~OCIContainerInitializedTest() override
+    {
+        EXPECT_CALL(dobbymock, unregisterListener(5))
+            .WillOnce(::testing::Return());
+
+        plugin->Deinitialize(&service);
+    }
+};
+
+
+
+TEST_F(OCIContainerTest, RegisteredMethods){
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("listContainers")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getContainerState")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getContainerInfo")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("startContainer")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("startContainerFromDobbySpec")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("stopContainer")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("pauseContainer")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("resumeContainer")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("executeCommand")));
+ }
+
+ TEST_F(OCIContainerInitializedTest, listContainersTest){
+
+    std::list<std::pair<int32_t, std::string>> containerslist = {{91, "com.bskyb.epgui"},{94, "Netflix"}};
+    EXPECT_CALL(dobbymock, listContainers())
+        .WillOnce(::testing::Return(containerslist));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("listContainers"), _T("{}"), response));
+    EXPECT_EQ(response, string("{\"containers\":[{\"Descriptor\":91,\"Id\":\"com.bskyb.epgui\"},{\"Descriptor\":94,\"Id\":\"Netflix\"}],\"success\":true}"));
+}
+
+ TEST_F(OCIContainerInitializedTest, getContainerStateTest){
+
+    std::list<std::pair<int32_t, std::string>> containerslist = {{91, "com.bskyb.epgui"},{94, "Netflix"}};
+    EXPECT_CALL(dobbymock, listContainers())
+        .WillOnce(::testing::Return(containerslist));
+
+    EXPECT_CALL(dobbymock, getContainerState(91))
+        .WillOnce(::testing::Return(2));
```

Comment:
Please use constants/enums intead of the magic numbers.

Reply:
I've made the changes if you could review them when you have a chance please.